### ISSUE TITLE
Cannot serve when using npm workspaces

### ISSUE THREAD

**Issue by hollandThomas** [tajo/ladle#233]
**Describe the bug**

In a setup that utilizes `npm workspaces` (e.g. to have a separate package that is a Remix app), ladle (a.k.a. vite) is unable to resolve `*.stories.tsx` files correctly.

**Reproduction**

Not sure if/how it's possible to create a `npm workspaces` setup with Stackblitz, hence follow these steps:

1. `npm init` a fresh npm project
2.  add a `workspaces` key to `package.json`

     ```json
      "workspaces": [
        "frontend"
      ],
      ```
3. `npx create-remix@latest`
     `./frontend`
     `Just the basics`
     `Remix App Server`
     `TypeScript`
     `Y`
4. In `frontend/package.json`, give it a `name` and add `ladle` to `scripts`

    ```json
    "name": "frontend",
    "scripts": {
      "build": "remix build",
      "dev": "remix dev",
      "start": "remix-serve build",
      "ladle": "DEBUG=ladle* npx ladle serve"
    },
    ```
5. Create a file `frontend/app/components/my-components.stories.tsx`

   ```tsx
   export const Button = () => <button>My Button</button>;
   export const HeaderOne = () => <h1>Header</h1>;
   ```
6. Create `frontend/.ladle/config.mjs`

    ```js
    export default {
      stories: './app/components/**/*.stories.{js,jsx,ts,tsx}',
    };
    ```
7. `npm i -w frontend @ladle/react`
8. `npm run -w frontend ladle`

**Vite overlay error**

> [plugin:vite:import-analysis] Cannot import non-asset file /../../../../../frontend/app/components/my-components.stories.tsx which is inside /public.JS/CSS files inside /public are copied as-is on build and can only be referenced via <script src> or <link href> in html.

**Output from**

```sh
DEBUG=ladle* npx ladle serve
```

```sh
 ladle:cli Starting serve command +0ms
  ladle:cli CLI theme: undefined +2ms
  ladle:cli CLI stories: undefined +0ms
  ladle:cli CLI port: undefined +0ms
  ladle:cli CLI out: undefined +0ms
  ladle:cli Custom config found: {
  ladle:cli   "stories": "app/components/**/*.stories.{js,jsx,ts,tsx}"
  ladle:cli } +20ms
  ladle:cli Final config:
  ladle:cli {
  ladle:cli   "stories": "app/components/**/*.stories.{js,jsx,ts,tsx}",
  ladle:cli   "defaultStory": "",
  ladle:cli   "port": 61000,
  ladle:cli   "previewPort": 8080,
  ladle:cli   "outDir": "build",
  ladle:cli   "addons": {
  ladle:cli     "control": {
  ladle:cli       "enabled": true,
  ladle:cli       "defaultState": {}
  ladle:cli     },
  ladle:cli     "theme": {
  ladle:cli       "enabled": true,
  ladle:cli       "defaultState": "light"
  ladle:cli     },
  ladle:cli     "mode": {
  ladle:cli       "enabled": true,
  ladle:cli       "defaultState": "full"
  ladle:cli     },
  ladle:cli     "rtl": {
  ladle:cli       "enabled": true,
  ladle:cli       "defaultState": false
  ladle:cli     },
  ladle:cli     "source": {
  ladle:cli       "enabled": true,
  ladle:cli       "defaultState": false
  ladle:cli     },
  ladle:cli     "a11y": {
  ladle:cli       "enabled": true
  ladle:cli     },
  ladle:cli     "ladle": {
  ladle:cli       "enabled": true
  ladle:cli     },
  ladle:cli     "width": {
  ladle:cli       "enabled": true,
  ladle:cli       "options": {
  ladle:cli         "xsmall": 414,
  ladle:cli         "small": 640,
  ladle:cli         "medium": 768,
  ladle:cli         "large": 1024
  ladle:cli       },
  ladle:cli       "defaultState": 0
  ladle:cli     }
  ladle:cli   }
  ladle:cli } +1ms
  ladle:cli Port set to: 61000 +14ms

   ╭────────────────────────────────────────────────────╮
   │                                                    │
   │   🥄 Ladle.dev served at http://localhost:61000/   │
   │                                                    │
   ╰────────────────────────────────────────────────────╯

  ladle:vite Parsing app/components/my-components.stories.tsx +0ms
  ladle:vite Parsed data for app/components/my-components.stories.tsx: +33ms
  ladle:vite {
  ladle:vite   entry: 'app/components/my-components.stories.tsx',
  ladle:vite   stories: [
  ladle:vite     {
  ladle:vite       storyId: 'my-components--button',
  ladle:vite       componentName: 'my$components$$button',
  ladle:vite       namedExport: 'Button',
  ladle:vite       locStart: 1,
  ladle:vite       locEnd: 1
  ladle:vite     },
  ladle:vite     {
  ladle:vite       storyId: 'my-components--header-one',
  ladle:vite       componentName: 'my$components$$header$one',
  ladle:vite       namedExport: 'HeaderOne',
  ladle:vite       locStart: 2,
  ladle:vite       locEnd: 2
  ladle:vite     }
  ladle:vite   ],
  ladle:vite   exportDefaultProps: { title: undefined, meta: undefined },
  ladle:vite   namedExportToMeta: {},
  ladle:vite   namedExportToStoryName: {},
  ladle:vite   storyParams: {},
  ladle:vite   storySource: 'export const Button = () => <button>My Button</button>;\n' +
  ladle:vite     'export const HeaderOne = () => <h1>Header</h1>;\n',
  ladle:vite   fileId: 'my-components'
  ladle:vite } +0ms
  ladle:vite transforming: virtual:generated-list +0ms
  ladle:vite Initial generation of the list +0ms
  ladle:vite Parsing app/components/my-components.stories.tsx +624ms
  ladle:vite Parsed data for app/components/my-components.stories.tsx: +2ms
  ladle:vite {
  ladle:vite   entry: 'app/components/my-components.stories.tsx',
  ladle:vite   stories: [
  ladle:vite     {
  ladle:vite       storyId: 'my-components--button',
  ladle:vite       componentName: 'my$components$$button',
  ladle:vite       namedExport: 'Button',
  ladle:vite       locStart: 1,
  ladle:vite       locEnd: 1
  ladle:vite     },
  ladle:vite     {
  ladle:vite       storyId: 'my-components--header-one',
  ladle:vite       componentName: 'my$components$$header$one',
  ladle:vite       namedExport: 'HeaderOne',
  ladle:vite       locStart: 2,
  ladle:vite       locEnd: 2
  ladle:vite     }
  ladle:vite   ],
  ladle:vite   exportDefaultProps: { title: undefined, meta: undefined },
  ladle:vite   namedExportToMeta: {},
  ladle:vite   namedExportToStoryName: {},
  ladle:vite   storyParams: {},
  ladle:vite   storySource: 'export const Button = () => <button>My Button</button>;\n' +
  ladle:vite     'export const HeaderOne = () => <h1>Header</h1>;\n',
  ladle:vite   fileId: 'my-components'
  ladle:vite } +0ms
  ladle:vite Returning default no-op Provider. +0ms
Cannot import non-asset file /../../../../../frontend/app/components/my-components.stories.tsx which is inside /public.JS/CSS files inside /public are copied as-is on build and can only be referenced via <script src> or <link href> in html.
Cannot import non-asset file /../../../../../frontend/app/components/my-components.stories.tsx which is inside /public.JS/CSS files inside /public are copied as-is on build and can only be referenced via <script src> or <link href> in html. (x2)
```

More about [troubleshooting](https://ladle.dev/docs/troubleshooting).

**Environment**

 - OS: macOS 12.4
 - Browser Chrome
 - Version 104
 - Node: 16.17.0
 - npm: 8.15.0

**Comment by tajo** [comment]
> Not sure if/how it's possible to create a npm workspaces setup with Stackblitz

It is (basically any node project really since it's a container) and it would be really helpful!

**Comment by tajo** [comment]
> JS/CSS files inside /public are copied as-is on build and can only be referenced via <script src> or in html

Is your code or stories inside of the `/public` folder?

**Comment by hollandThomas** [comment]
I don't really know how to replicate this in Stackblitz 😅 But here's a repo that you can clone
https://github.com/hollandThomas/ladle-reproduction

**Comment by tajo** [comment]
That's a good repro. I am really confused where the error's coming from, pretty concerning. Will look into it more this week.

**Comment by tajo** [comment]
fixed with `@ladle/react@2.4.3`
